test/augmenters/test_geometric.py

```diff
@@ -3072,6 +3072,43 @@ def test_PerspectiveTransform():
         exterior_expected[:, 1] = ((exterior_expected[:, 1] - scale * 30) / (30 * (1-scale))) * 30
         observed.polygons[0].exterior_almost_equals(exterior_expected)
 
+
+
+    # ------------
+    # mode
+    # ------------
+    aug = iaa.PerspectiveTransform(cval=0, mode=ia.ALL)
+    assert isinstance(aug.mode, iap.Choice)
+    aug = iaa.PerspectiveTransform(cval=0, mode="replicate")
+    assert isinstance(aug.mode, iap.Deterministic)
+    assert aug.mode.value == "replicate"
+    aug = iaa.PerspectiveTransform(cval=0, mode=["replicate", "constant"])
+    assert isinstance(aug.mode, iap.Choice)
+    assert len(
+        aug.mode.a) == 2 and "replicate" in aug.mode.a and "constant" in aug.mode.a
+    aug = iaa.PerspectiveTransform(cval=0, mode=iap.Choice(["replicate", "constant"]))
+    assert isinstance(aug.mode, iap.Choice)
+    assert len(
+        aug.mode.a) == 2 and "replicate" in aug.mode.a and "constant" in aug.mode.a
+
+    # Check new values
+    img = np.ones((256, 256, 3), dtype=np.uint8) * 255
+    aug = iaa.PerspectiveTransform(scale=0.001, mode='replicate', cval=0,
+                                   random_state=np.random.RandomState(seed=31))
+    img = aug.augment_image(img)
```

Comment:
You overwrite the input image here, which you use for the following tests. Something like `img_aug = aug.augment_image(img)` should fix that.